src/dawn/SIR/SIR.cpp

```diff
@@ -174,4 +174,137 @@ std::shared_ptr<sir::StencilCall> sir::StencilCall::clone() const {
   return call;
 }
 
+bool SIR::operator==(const SIR& rhs) const {
+  bool retval = true;
+
+  if((Stencils.size() == rhs.Stencils.size()) &&
+     (StencilFunctions.size() == rhs.StencilFunctions.size())) {
+
+    if(Stencils.size() > 0) {
+      retval &= std::equal(Stencils.begin(), Stencils.end(), rhs.Stencils.begin(),
+                           sharedpointer_comparison<sir::Stencil>);
+    }
+
+    if(StencilFunctions.size() > 0) {
+      retval &=
+          std::equal(StencilFunctions.begin(), StencilFunctions.end(), rhs.StencilFunctions.begin(),
+                     sharedpointer_comparison<sir::StencilFunction>);
+    }
+
+    if(GlobalVariableMap != nullptr && rhs.GlobalVariableMap != nullptr) {
+      if(GlobalVariableMap->size() == rhs.GlobalVariableMap->size()) {
+        if(GlobalVariableMap->size() > 0) {
+          retval &=
+              pointerMapComparison(*(GlobalVariableMap.get()), *(rhs.GlobalVariableMap.get()));
+        }
+      } else {
+        return false;
+      }
+    } else {
+      return GlobalVariableMap == rhs.GlobalVariableMap;
+    }
+
+    return retval;
+  }
+  return false;
+}
+bool SIR::operator!=(const SIR& rhs) const { return !(*this == rhs); }
+
+bool sir::Stencil::operator==(const sir::Stencil& rhs) const {
+  bool retval = true;
+  if(Fields.size() != rhs.Fields.size()) {
+    return false;
+  }
+  retval &= (Name == rhs.Name);
+  retval &= (Attributes == rhs.Attributes);
+  if(StencilDescAst != nullptr && rhs.StencilDescAst != nullptr) {
+    if(StencilDescAst->getRoot() != nullptr && rhs.StencilDescAst->getRoot() != nullptr) {
+      retval &= StencilDescAst->getRoot().get()->equals(rhs.StencilDescAst->getRoot().get());
+    } else {
+      return StencilDescAst->getRoot() == rhs.StencilDescAst->getRoot();
```

Comment:
Remove?

Reply:
Again, same argument as for the GlobalVariableMap, if somehow both StencilDescAsts are initialized with a nullptr, we need this for equality (and the if before makes it s.t. get and equals work properly

Reply:
Ok